compiler/rustc_ast_lowering/src/expr.rs

```diff
@@ -958,6 +960,7 @@ impl<'hir> LoweringContext<'_, 'hir> {
         binder: &ClosureBinder,
         capture_clause: CaptureBy,
         closure_id: NodeId,
+        closure_hir_id: hir::HirId,
```

Comment:
is this the same id as the closure_id below (but lowered node id -> hir id?) it feels weird and redundant to pass both in, unless one is distinct

Reply:
yea it is. I copied this duplicate passing from `lower_expr_coroutine_closure`. I think we can move to using `HirId` everywhere, but it's a bit of an invasive refactoring

Reply:
ok, let's not do that here